src/tests/Microsoft.PowerFx.Core.Tests/TexlTests.cs

```diff
@@ -117,5 +147,2211 @@ public void TexlDateTableFunctions_Negative(string expression)
+        public void TexlFunctionTypeSemanticsReplace_ConsistentOneColumnTableResult(string script, string expectedType)
+        {
+            var symbol = new SymbolTable();
+            symbol.AddVariable("T", new TableType(TestUtils.DT("*[Name:s]")));
+            symbol.AddVariable("T2", new TableType(TestUtils.DT("*[Start:n]")));
+            symbol.AddVariable("T3", new TableType(TestUtils.DT("*[Count:n]")));
+            symbol.AddVariable("TX", new TableType(TestUtils.DT("*[Replacement:s]")));
+
+            TestSimpleBindingSuccess(
+                script,
+                TestUtils.DT(expectedType),
+                symbol,
+                Features.ConsistentOneColumnTableResult);
+        }
+
+        [Fact]
+        public void TexlFunctionTypeSemanticsInt()
+        {
+            var symbol = new SymbolTable();
+            symbol.AddVariable("T", new TableType(TestUtils.DT("*[A:n]")));
+
+            TestSimpleBindingSuccess(
+                "Int(1234)",
+                DType.Number);
+
+            TestSimpleBindingSuccess(
+                "Int(T)",
+                TestUtils.DT("*[A:n]"),
+                symbol);
+        }
+
+        [Theory]
+        [InlineData("Int(\"3\")", "n", null)]
+        [InlineData("Int(true)", "n", null)]
+        [InlineData("Int(T)", "*[Booleans:n]", "*[Booleans:b]")]
+        [InlineData("Int(T)", "*[Strings:n]", "*[Strings:s]")]
+        [InlineData("Int([true, false, true])", "*[Value:n]", null)]
+        [InlineData("Int([\"5\", \"6\"])", "*[Value:n]", null)]
+        public void TexlFunctionTypeSemanticsIntWithCoercion(string script, string expectedType, string typedGlobal)
+        {
+            var symbol = new SymbolTable();
+            
+            if (typedGlobal != null)
+            {
+                symbol.AddVariable("T", new TableType(TestUtils.DT(typedGlobal)));
+                TestSimpleBindingSuccess(
+                    script,
+                    TestUtils.DT(expectedType),
+                    symbol);
+            }
+            else
+            {
+                TestSimpleBindingSuccess(script, TestUtils.DT(expectedType));
+            }
+        }
+
+        [Fact]
+        public void TexlFunctionTypeSemanticsTruncOneParam()
+        {
+            TestSimpleBindingSuccess(
+                "Trunc(1234)",
+                DType.Number);
+
+            var symbol = new SymbolTable();
+            symbol.AddVariable("T", new TableType(TestUtils.DT("*[A:n]")));
+            TestSimpleBindingSuccess(
+                "Trunc(T)",
+                TestUtils.DT("*[A:n]"),
+                symbol);
+        }
+
+        [Fact]
+        public void TexlFunctionTypeSemanticsTruncOneParam_ConsistentOneColumnTableResult()
+        {
+            var symbol = new SymbolTable();
+            symbol.AddVariable("T", new TableType(TestUtils.DT("*[A:n]")));
+            TestSimpleBindingSuccess(
+                "Trunc(T)",
+                TestUtils.DT("*[A:n]"),
```

Comment:
>*[A:n] [](http://example.com/codeflow?start=30&length=6)

If we're using the ConsistentOneColumnTableResult flag, it should be `*[Value:n]`

Reply:
The current code returns the same return type as arg0, is that not expected behavior?

Reply:
No, the current code is wrong (we found a bug, great!) The "ConsistentOneColumnTableResult" flag will change the schema of all functions that return a single-column table (tabular overloads of scalar functions) to always have a *[Value:<type>] schema.

Reply:
I have filed issue https://github.com/microsoft/Power-Fx/issues/709 for this bug.